vault/core_test.go

```diff
@@ -186,6 +187,155 @@ func TestCore_Unseal_Single(t *testing.T) {
 	}
 }
 
+func TestCore_Unseal_DataVersion(t *testing.T) {
+	// Ignore version pre-release
+	version.VersionPrerelease = ""
+
+	// Case where local-version is greater than data-version
+	version.Version = "0.7.0"
+	c, keys, _ := TestCoreDataVersion(t, "0.6.0")
+
+	for _, key := range keys {
+		_, err := TestCoreUnseal(c, key)
+		if err != nil {
+			t.Fatalf("err: %v", err)
+		}
+	}
+
+	sealed, err := c.Sealed()
+	if err != nil {
+		t.Fatalf("err: %v", err)
+	}
+	if sealed {
+		t.Fatalf("should not be sealed")
+	}
+
+	// Case where local-version is less than data-version
+	version.Version = "0.6.0"
+	c, keys, _ = TestCoreDataVersion(t, "0.7.0")
+
+	for i, key := range keys {
+		_, err := TestCoreUnseal(c, key)
```

Comment:
Same comment as above; you should only get an error when the downgrade happens, so rather than playing with the length of the keys slice, simply call TestCoreUnsealed until you get `true` or an error that isn't nil and break; then you can check that it's still sealed (and if you break at that point, you can verify that the error is not nil, too).